Mathlib/RingTheory/Polynomial/Resultant/Basic.lean

```diff
@@ -43,44 +44,54 @@ namespace Polynomial
 
 section sylvester
 
-variable {R : Type*} [Semiring R]
+variable {R S : Type*} [Semiring R] [Semiring S]
 
 /-- The Sylvester matrix of two polynomials `f` and `g` of degrees `m` and `n` respectively is a
-`(n+m) × (n+m)` matrix with the coefficients of `f` and `g` arranged in a specific way. Here, `m`
+`(m+n) × (m+n)` matrix with the coefficients of `f` and `g` arranged in a specific way. Here, `m`
 and `n` are free variables, not necessarily equal to the actual degrees of the polynomials `f` and
 `g`. -/
-def sylvester (f g : R[X]) (m n : ℕ) : Matrix (Fin (n + m)) (Fin (n + m)) R :=
+def sylvester (f g : R[X]) (m n : ℕ) : Matrix (Fin (m + n)) (Fin (m + n)) R :=
```

Comment:
My first instinct would be to define:
```lean
def sylvester (f g : R[X]) (m n : ℕ) : Matrix (Fin m ⊕ Fin n) (Fin m ⊕ Fin n) R :=
```
Since you're adjusting this definition, do you think this is worth considering?

Reply:
The "natural definition" would be `Matrix (Fin (m + n)) (Fin m ⊕ Fin n) R` but we need a square matrix to take determinant so we need to make an arbitrary choice. I think the map from `Fin (m + n)` to `Fin m ⊕ Fin n` has more API so this choice is more easy.